pkg/asset/installconfig/gcp/client.go

```diff
@@ -109,6 +110,40 @@ func (c *Client) GetPublicDomains(ctx context.Context, project string) ([]string
 	return publicZones, nil
 }
 
+// GetDNSZoneByName returns a DNS zone matching the `zoneName` if the DNS zone exists
+// and can be seen (correct permissions for a private zone) in the project.
+func (c *Client) GetDNSZoneByName(ctx context.Context, defaultProject, providedZoneName string) (*dns.ManagedZone, error) {
+	ctx, cancel := context.WithTimeout(context.TODO(), 1*time.Minute)
+	defer cancel()
+
+	project, zoneName, err := ParseZone(defaultProject, providedZoneName)
```

Comment:
If we remove the defaultProject argument, this would need to change
```
	project, zoneName, err := ParseZoneName(providedZoneName)
	if err != nil {
		return nil, err
	}
	if project == nil or project == "" {
		project = defaultProject
	}
```